jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/model/CodeModernizerSessionContext.kt

```diff
@@ -225,6 +244,84 @@ data class CodeModernizerSessionContext(
+            val pomcp = "-Dmdep.copyPom=true"
+            val parentpom = "-Dmdep.addParentPoms=true"
+            val commandlist = mutableListOf<String>()
+            val explicitenabled = mutableListOf<String>()
+            commandlist.add(goalcp)
+            commandlist.add(outputDirectory)
+            commandlist.add(repolay)
+            commandlist.add(pomcp)
+            commandlist.add(parentpom)
+            try {
+                val params = MavenRunnerParameters(
+                    false,
+                    sourceFolder.absolutePath,
+                    null,
+                    commandlist,
+                    explicitenabled,
+                    null
+                )
+
+                // Create MavenRunnerParametersMavenRunnerParameters
+                val mvnrunner = MavenRunner.getInstance(project)
+                val transformMvnRunner = TransformMavenRunner(project)
+                val mvnsettings = mvnrunner.settings
+                val createdDependencies = TransformRunnable()
+                runInEdt {
+                    try {
+                        transformMvnRunner.run(params, mvnsettings, createdDependencies)
+                    } catch (t: Throwable) {
+                        createdDependencies.exitCode(Integer.MIN_VALUE) // to stop looking for the exitCode
+                        LOG.error { t.message.toString() }
+                        CodetransformTelemetry.mvnBuildFailed(
+                            codeTransformSessionId = CodeTransformTelemetryState.instance.getSessionId(),
+                            codeTransformMavenBuildCommand = CodeTransformMavenBuildCommand.IDEBundledMaven,
+                            reason = t.message
+                        )
+                    }
+                }
+                while (createdDependencies.isComplete() == null) {
+                    // waiting mavenrunner building
+                    sleep(50)
+                }
+                if (createdDependencies.isComplete() == 0) {
+                    LOG.warn { "IntelliJ bundled Maven executed successfully" }
+                } else if (createdDependencies.isComplete() != Integer.MIN_VALUE) {
+                    val error = "The exitCode should be 0 while it was ${createdDependencies.isComplete()}"
+                    LOG.error { error }
+                    CodetransformTelemetry.mvnBuildFailed(
+                        codeTransformSessionId = CodeTransformTelemetryState.instance.getSessionId(),
+                        codeTransformMavenBuildCommand = CodeTransformMavenBuildCommand.IDEBundledMaven,
+                        reason = error
+                    )
+                    return null
+                } else {
+                    // when exit code is MIN_VALUE
+                    // return null
+                    return null
+                }
+            } catch (t: Throwable) {
+                LOG.error { t.message.toString() }
```

Comment:
shouldn't we give the user a formatted message explaining why there was a failure?

Reply:
We will give customers a notification without failure reasons in `CodeModernizerSession`  when all the maven commands fail